utils/wasm-gen/src/config/syscalls/injection.rs

```diff
@@ -0,0 +1,121 @@
+// This file is part of Gear.
+
+// Copyright (C) 2021-2023 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! Entities describing possible injection types for each sys-call.
+//! These entities allows to configure which sys-calls to insert into
+//! code section of wasm module and which ones to simply import.
+//!
+//! Types here are used to create [`crate::SysCallsConfig`].
+
+use crate::InvocableSysCall;
+
+use gear_wasm_instrument::syscalls::SysCallName;
+use std::{collections::HashMap, ops::RangeInclusive};
+
+/// This enum defines how the sys-call should be injected into wasm module.
+#[derive(Debug, Clone)]
+pub enum SysCallInjectionType {
+    /// Don't modify wasm module at all.
+    None,
+    /// Sys-call import will be injected into import section of wasm module,
+    /// but the `wasm-gen` generators will not call that sys-call.
```

Comment:
can you add please that this variant is needed to serve precise syscalls generation and etc.

Reply:
done in 0612f9d084ed1d50f7fdfad3e640c5157c225fd7